capsules/extra/src/can.rs

```diff
@@ -461,36 +460,20 @@ impl<'a, Can: can::Can> can::ReceiveClient<{ can::STANDARD_CAN_PACKET_SIZE }>
                                         buffer_ref
                                             .mut_enter(|user_buffer| {
```

Comment:
@lschuermann Do we need a different API here then? How does a capsule get an object of the wrapper type?

Reply:
Given that our grant API yields a `ReadOnlyProcessBuffer` reference, I was thinking that we can just wrap this type into `$ThisPRsTypeName(ReadOnlyProcessBuffer)` that then exposes the APIs proposed here.

We'd rely on capsules correctly wrapping the process buffers prior to interacting with them, but that's similar to how we currently trust capsules to actually enforce their driver API contracts.

Reply:
That makes sense to me.

Reply:
Ok, so if you want to use the ProcessBufferSlice as a Buffer, wrap it first into a Buffer type?